core/partitioning/partitioning.cpp

```diff
@@ -0,0 +1,232 @@
+    // kTorch segments
+    if (segmented_blocks[use_info.produce_id].target() == SegmentedBlock::kTensorRT && !use_info.torch_use_id.empty()) {
+      int first_torch_id = use_info.torch_use_id.front();
+      if (!updated_segments.count(first_torch_id)) {
+        auto new_torch_block = injectNodesForNonTensorInputs(segmented_blocks[first_torch_id]);
+        segmented_blocks[first_torch_id] = new_torch_block;
+        updated_segments.insert(first_torch_id);
+      }
+    } else {
+      // KTensorRT segments always need to inject nodes for the nonTensor inputs
+      for (int i : use_info.tensorrt_use_id) {
+        if (!updated_segments.count(i)) {
+          auto new_seg_block = injectNodesForNonTensorInputs(segmented_blocks[i]);
+          segmented_blocks[i] = new_seg_block;
+          updated_segments.insert(i);
+        }
+      }
+    }
+  }
+  return;
+}
+
+void registerSegmentsOutputs(PartitionedGraph& segmented_blocks, std::shared_ptr<torch::jit::Graph> g) {
+  // find the corresponding raw values in original global graph for this segmented block's inputs/outputs
+  std::set<torch::jit::Value*> input_values;
+  for (auto& seg_block : segmented_blocks) {
+    for (auto& input : seg_block.raw_inputs()) {
+      input_values.insert(input);
+    }
+  }
+
+  for (auto& graph_output : g->outputs()) {
+    input_values.insert(graph_output);
+  }
+
+  // should be careful here because some in-place operations don't return any values, there is no output for this kind
+  // of segment identify the output for each mini-graph by checking if any value in this graph is used later we
+  // shouldn't register nonTensor output for TensorRT segments
+  for (auto& seg_block : segmented_blocks) {
+    for (auto& mini_graph_input : input_values) {
+      if (std::find(seg_block.raw_inputs().begin(), seg_block.raw_inputs().end(), mini_graph_input) ==
+              seg_block.raw_inputs().end() &&
+          seg_block.contain_raw_value(mini_graph_input)) {
+        if (!isTensorOrTensorList(mini_graph_input) && seg_block.target() == SegmentedBlock::kTensorRT)
+          continue;
+        seg_block.registerOutput(mini_graph_input);
+      }
+    }
+    // if no output, then register the last node's output as current graph's output
+    if (seg_block.raw_outputs().empty()) {
+      // for Torch segments, register input as output
+      if (seg_block.target() == SegmentedBlock::kTorch) {
+        seg_block.registerOutput(seg_block.raw_inputs()[0]);
+      } else {
+        // for TensorRT segments, register last nonInput Tensor outputs
+        for (int i = seg_block.raw_nodes().size() - 1; i >= 0; --i) {
+          for (auto node_output : seg_block.raw_nodes()[i]->outputs()) {
+            if (isTensorOrTensorList(node_output))
+              seg_block.registerOutput(node_output);
+          }
+          if (!seg_block.raw_outputs().empty())
+            break;
+        }
+      }
+    }
+  }
+  // erase segments which still have no output

```

Comment:
You can just use the DeadCodeElimination Pass to do this

Reply:
DeadCodeElimination seems to erase dead code in a graph, but here we erase the useless SegmentedBlock. Maybe we can add that pass for every SegmentedBlock to erase dead code?

Reply:
Yeah we should do that